torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3085,8 +3078,39 @@ def _wait_for_post_backward(self) -> None:
         # A backward pass is done, clean up below.
         self._exec_order_data.reset()
 
+        def _catch_all_reshard(fsdp_module: FullyShardedDataParallel) -> None:
+            """
+            Reshards full parameters that may have not been resharded in
+            post_backward_hook. This can happen when an FSDP module's output
+            is used in forward so its pre-backward fires unsharding the param,
+            but post-backward does not fire since the output was not ultimately
+            used in loss computation so FSDP parameter did not get a gradient.
+            """
+            try:
```

Comment:
I am curious what is the motivation for the `try: ... except: ...`?

It looks like it is to print any error from the backward pass?

Reply:
yes, exactly. Ideally, we should make sure none of our calls in `backward` can throw and use `p_assert` before calling any function to validate inputs, but that may require some more combing through all the code that runs in backwards, so just wrapping it this way for now. added a comment to indicate so.